test/unit/zendesk-uploader/client-wrapper.js

```diff
@@ -146,13 +146,48 @@ describe('ZendeskClientWrapper', () => {
             });
         });
 
-        it('should reject the promise if the request fails after 5 retries', () => {
+        it('should retry after twice as long as the last retry for the request when there are subsequent 500 errors', () => {
+            const stub = sandbox.stub(this.zendeskStub.articles, 'create').yields({statusCode: 503});
+            stub.onCall(3).yields(null, null, {});
+
+            let clock = sandbox.useFakeTimers();
+            let createArticle = this.zendeskClient.articles.create(123, {});
+
+            clock.tick(500);
+            expect(stub).to.be.calledTwice;
+
+            clock.tick(1000);
+            expect(stub).to.be.calledThrice;
+
+            clock.tick(2000);
+            return createArticle.then(() => {
+                expect(stub.callCount).to.be.equal(4);
+            });
+        });
+
+        it('should not wait longer than 8000ms on subsequent retries due to 500 errors', () => {
+            const stub = sandbox.stub(this.zendeskStub.articles, 'create').yields({statusCode: 503});
+            stub.onCall(6).yields(null, null, {});
+
+            let clock = sandbox.useFakeTimers();
+            let createArticle = this.zendeskClient.articles.create(123, {});
+
+            clock.tick(500 + 1000 + 2000 + 4000 + 8000);
+            expect(stub.callCount).to.be.equal(6);
```

Comment:
This one is checking that it's called 6 times, then ticking another 8000, then checks that it was called 7 times. I don't think this one needs any extra checks because the test above is already checking that the times are incremented correctly.